### ISSUE TITLE
`cp : Access to the path 'C:\etc\kube-flannel\net-conf.json' is denied in flannel hostprocess start.ps1

### ISSUE THREAD

**Issue by yuwei1chen** [kubernetes-sigs/sig-windows-tools#316]
**Describe the bug**
start.ps1 reports below error when kube-flannel-ds-windows-amd64 starts on windows node
`cp : Access to the path 'C:\etc\kube-flannel\net-conf.json' is denied
`
It seems that the flannel hostprocess container is having access issue? But I see there is a copy-item step copying cni bin in the script which seems to work fine

**To Reproduce**
follow the steps here: https://github.com/kubernetes-sigs/sig-windows-tools/blob/master/guides/flannel.md to deploy flannel for windows worker nodes.

**Expected behavior**
 'C:\etc\kube-flannel\net-conf.json' is copied successfully and flannel is started.

**Kubernetes (please complete the following information):**
 - Windows Server version: 
 - wmic os get Caption
Caption
Microsoft Windows Server 2019 Datacenter

Version
10.0.17763

BuildNumber
17763

OSArchitecture
64-bit
 - Kubernetes Version: v1.27.1
 - CNI: containerd

**Additional context**

**Comment by Mik4sa** [comment]
The images here should be outdated. You better build your own and use these
If you want, you can use this image for testing before building your own: https://hub.docker.com/layers/mik4sa/flannel/v0.21.5-hostprocess/images/sha256-f4f1f2e7ed66e54f5fc9a6dde4980739afe666fc33058ddce1b280a311f1b85c?context=explore

But note! These are subject to change or removed anytime. So switch to your own images once verified that this works

**Comment by yuwei1chen** [comment]
> The images here should be outdated. You better build your own and use these If you want, you can use this image for testing before building your own: https://hub.docker.com/layers/mik4sa/flannel/v0.21.5-hostprocess/images/sha256-f4f1f2e7ed66e54f5fc9a6dde4980739afe666fc33058ddce1b280a311f1b85c?context=explore
> 
> But note! These are subject to change or removed anytime. So switch to your own images once verified that this works

@Mik4sa Hi, thanks for your image. This issue is gone none. But I see new error in the log:

```
Starting flannel
I0518 18:15:35.949732   19344 main.go:211] CLI flags config: {etcdEndpoints:http://127.0.0.1:4001,http://127.0.0.1:2379 etcdPrefix:/coreos.com/network etcdKeyfile: etcdCertfile: etcdCAFile: etcdUsername: etcdPassword: version:false kubeSubnetMgr:true kubeApiUrl: kubeAnnotationPrefix:flannel.alpha.coreos.com kubeConfigFile:C:\hpc\/flannel-config-file/kubeconfig.conf iface:[10.165.129.117] ifaceRegex:[] ipMasq:false ifaceCanReach: subnetFile:/run/flannel/subnet.env publicIP: publicIPv6: subnetLeaseRenewMargin:60 healthzIP:0.0.0.0 healthzPort:0 iptablesResyncSeconds:5 iptablesForwardRules:true netConfPath:/etc/kube-flannel/net-conf.json setNodeNetworkUnavailable:true useMultiClusterCidr:false}
I0518 18:15:36.742405   19344 kube.go:485] Starting kube subnet manager
I0518 18:15:36.742405   19344 kube.go:144] Waiting 10m0s for node controller to sync
I0518 18:15:36.937224   19344 kube.go:506] Creating the node lease for IPv4. This is the n.Spec.PodCIDRs: [10.244.1.0/24]
I0518 18:15:36.937793   19344 kube.go:506] Creating the node lease for IPv4. This is the n.Spec.PodCIDRs: [10.244.0.0/24]
I0518 18:15:37.756627   19344 kube.go:151] Node controller sync successful
I0518 18:15:37.756627   19344 main.go:231] Created subnet manager: Kubernetes Subnet Manager - jf51101cp-hv
I0518 18:15:37.756627   19344 main.go:234] Installing signal handlers
I0518 18:15:37.756627   19344 main.go:542] Found network config - Backend type: vxlan
I0518 18:15:37.757168   19344 match.go:73] Searching for interface using 10.165.129.117
I0518 18:15:37.762831   19344 match.go:259] Using interface with name vEthernet (vSwitch) and address 10.165.129.117
I0518 18:15:37.762831   19344 match.go:281] Defaulting external address to interface address (10.165.129.117)
I0518 18:15:37.763551   19344 vxlan_windows.go:125] VXLAN config: Name=flannel.4096 MacPrefix=0E-2A VNI=4096 Port=4789 GBP=false DirectRouting=false
time="2023-05-18T18:15:37-07:00" level=info msg="HCN feature check" supportedFeatures="{Acl:{AclAddressLists:true AclNoHostRulePriority:true AclPortRanges:true AclRuleId:true} Api:{V1:true V2:true} RemoteSubnet:true HostRoute:true DSR:true Slash32EndpointPrefixes:true AclSupportForProtocol252:false SessionAffinity:false IPv6DualStack:false SetPolicy:false VxlanPort:false L4Proxy:true L4WfpProxy:false TierAcl:false NetworkACL:false NestedIpSet:false}" version="{Major:9 Minor:5}"
I0518 18:15:37.775874   19344 device_windows.go:117] Attempting to create HostComputeNetwork &{ flannel.4096 Overlay [] {[]} { [] [] []} [{Static [{10.244.1.0/24 [[123 34 84 121 112 101 34 58 34 86 83 73 68 34 44 34 83 101 116 116 105 110 103 115 34 58 123 34 73 115 111 108 97 116 105 111 110 73 100 34 58 52 48 57 54 125 125]] [{10.244.1.1 0.0.0.0/0 0}]}]}] 8 {<nil> {[] []    }} {2 0}}
E0518 18:15:37.781045   19344 main.go:334] Error registering network: failed to create HostComputeNetwork flannel.4096: hcnCreateNetwork failed in Win32: An adapter was not found. (0x803b0006) {"Success":false,"Error":"An adapter was not found. ","ErrorCode":2151350278}
I0518 18:15:37.781045   19344 main.go:522] Stopping shutdownHandler...
W0518 18:15:37.782698   19344 reflector.go:347] pkg/subnet/kube/kube.go:486: watch of *v1.Node ended with: an error on the server ("unable to decode an event from the watch stream: context canceled") has prevented the request from succeeding
```

**Comment by Mik4sa** [comment]
I never had this issue so I can only guess. Was the correct network adapter chosen?

**Comment by yuwei1chen** [comment]
> I never had this issue so I can only guess. Was the correct network adapter chosen?

Yes. "interface with name vEthernet (vSwitch) and address 10.165.129.117" is the correct adapter.

**Comment by yuwei1chen** [comment]
> > I never had this issue so I can only guess. Was the correct network adapter chosen?
> 
> Yes. "interface with name vEthernet (vSwitch) and address 10.165.129.117" is the correct adapter.

@Mik4sa Should I create a new issue for this?

**Comment by Mik4sa** [comment]
Yeah, I think so :)

**Comment by jsturtevant** [comment]
@yuwei1chen yes if you are still experiencing an issue a new issue would be helpful thanks!
This issue was resolved by #297 
/close
